packages/tupaia-web/src/features/Map/MapTableModal/MapTableModal.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import styled from 'styled-components';
+import DownloadIcon from '@material-ui/icons/GetApp';
+import MuiIconButton from '@material-ui/core/IconButton';
+import { FlexColumn } from '@tupaia/ui-components';
+import { Modal } from '../../../components';
+import { MapTable } from '@tupaia/ui-map-components';
+import { useParams } from 'react-router';
+import { useEntityAncestors } from '../../../api/queries';
+
+const Wrapper = styled(FlexColumn)`
+  justify-content: flex-start;
+`;
+
+const Title = styled.div.attrs({
+  variant: 'h2',
+})`
+  font-size: 1.25rem;
+  line-height: 1.4;
+  font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  margin-bottom: 1.5rem;
+`;
+
+const IconButton = styled(MuiIconButton)`
+  margin-bottom: 1.5rem;
+`;
+
+const TitleWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  margin-bottom: 3rem;
+  position: relative;
+  border-bottom: 1px solid rgba(255, 255, 255, 0.12);
+`;
+
+export const MapTableModal = ({
+  serieses,
+  selectedOverlay,
+  processedMeasureData,
+  setIsOpen,
+}: any) => {
+  const handleCloseModal = () => {
+    setIsOpen(false);
+  };
+
```

Comment:
I think that fetching the measure data and serieses etc, and processing them should be contained in this file. I know it sounds not right because we already do it in `DataVisualsLayer` but `react-query` is pretty smart, so if you query the same thing in to files, it only does the query once, so there won't be any unnecessary network requests.
So my recommendation would be to keep all the logic that the map table needs within this file, and make `DataVisualsLayer` totally unaware of the `MapTableModal`. Then, instead of rendering `MapTableModal` inside `DataVisualsLayer`, render it inside `DesktopMapOverlaySelector`. Then you won't need to pass around as many props, and can have the open/close logic in `DesktopMapOverlaySelector` as well.